### PR TITLE
Fix for duplicate ID in selected collection

### LINKED ISSUE #277: SELECT should not create a collection with duplicated items
Apparently, you can create a collection with duplicated items and some operations may be applied twice to those items. 

I was thinking on using this format:
{
      "func": "SELECT",
      "source": "DEFAULT"
      "collection": "DEFAULT",
      "property": "id",
      "value": "7grs", (ignored)
      "mode": "clean"
}

Using this format would allow filtering by any property:
{
      "func": "SELECT",
      "source": "DEFAULT"
      "collection": "DEFAULT",
      "property": "myCustomKey",
      "mode": "clean"
}

or output the result in a new collection:
{
      "func": "SELECT",
      "source": "DEFAULT"
      "collection": "NewCollection",
      "property": "id",
      "mode": "clean"
}

### PR DESCRIPTION
**Pull request by larkob**
I found that JavaScript does not prevent duplicate array entries in case of an array of objects. So, I had to add some lines so that widgets are ignored when their id is already in the collection.

Resolves #277

### CODE REVIEW

**Comment by ArnoldSmith86** [comment]
PR-SERVER-BOT: You can play around with it here: http://212.47.248.129:3280/pr-test (or any other room on that server)

File: client/js/widgets/button.js [hunk 1]
```diff
@@ -462,6 +462,10 @@ export class Button extends Widget {
           if([ 'add', 'set' ].indexOf(a.mode) == -1)
             problems.push(`Warning: Mode ${a.mode} interpreted as set.`);
           let c = (a.source == 'all' ? Array.from(widgets.values()) : collections[a.source]).filter(function(w) {
+            if (typeof collections[a.collection] === 'object') {
+              if(typeof collections[a.collection].find(o => o.id === w.p('id')) !== 'undefined')
```
Comment (ArnoldSmith86) [inline client/js/widgets/button.js:466]: While `o.id` probably works fine, it should be `o.p('id')` for consistency.

**Review by ArnoldSmith86 — APPROVED** [review]
Code looks fine. Example seems to work.

**Review by RaphaelAlvez — APPROVED** [review]
tested and approved

**Pull request merged**